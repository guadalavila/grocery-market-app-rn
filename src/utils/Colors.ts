export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
