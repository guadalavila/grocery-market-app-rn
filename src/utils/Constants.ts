export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
