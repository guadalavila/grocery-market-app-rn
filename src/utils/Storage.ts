import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
