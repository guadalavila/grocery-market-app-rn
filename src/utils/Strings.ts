export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
