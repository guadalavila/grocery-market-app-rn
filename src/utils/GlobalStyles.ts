import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
