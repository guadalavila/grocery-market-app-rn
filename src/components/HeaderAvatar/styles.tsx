import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        marginHorizontal: Sizes.M,
        marginVertical: Sizes.L,
    },
    title: {
        color: Colors.black,
        fontWeight: 'bold',
        fontSize: 30,
    },
    containerCart: {
        alignSelf: 'center',
        backgroundColor: Colors.primary,
        borderRadius: 20,
        padding: Sizes.S,
    },
    containerBadge: {
        backgroundColor: Colors.red,
        position: 'absolute',
        right: -6,
        top: -10,
        borderRadius: 18,
    },
    badge: {
        color: Colors.white,
        fontWeight: '600',
        paddingHorizontal: Sizes.XXS,
    },
});
