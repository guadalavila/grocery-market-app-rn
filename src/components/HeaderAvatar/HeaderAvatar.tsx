import React from 'react';
import { Image, Text, View } from 'react-native';
import { GlobalStyles } from '../../utils/GlobalStyles';
import { styles } from './styles';

const HeaderAvatar = () => {
    return (
        <View style={[GlobalStyles.row, GlobalStyles.justBtw, styles.container]}>
            <Text style={styles.title}>Grocery</Text>
            <Image source={require('../../../assets/images/user.png')} />
        </View>
    );
};

export default HeaderAvatar;
