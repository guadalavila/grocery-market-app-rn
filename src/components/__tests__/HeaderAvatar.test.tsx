import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import { Colors } from '../../utils/Colors';
import { Strings } from '../../utils/Strings';
import HeaderAvatar from '../HeaderAvatar/HeaderAvatar';
describe('test HeaderAvatar component', () => {
    const onPressMock = jest.fn();
    test('should match with snapchot', () => {
        const header = render(<HeaderAvatar onPress={onPressMock} />);
        expect(header).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<HeaderAvatar onPress={onPressMock} />);
        expect(screen).toBeTruthy();
    });

    test('should render title', () => {
        render(<HeaderAvatar onPress={onPressMock} />);
        expect(screen.getByText(Strings.appName)).toBeTruthy();
    });

    test('should render icon', () => {
        render(<HeaderAvatar onPress={onPressMock} />);
        expect(screen.getByTestId('icon-cart')).toBeTruthy();
        expect(screen.getByTestId('icon-cart').props.name).toBe('cart');
        expect(screen.getByTestId('icon-cart').props.size).toBe(22);
        expect(screen.getByTestId('icon-cart').props.color).toBe(Colors.white);
    });

    test('should render button', () => {
        render(<HeaderAvatar onPress={onPressMock} />);
        const btn = screen.getByTestId('button');
        expect(btn).toBeTruthy();
        fireEvent.press(btn);
        expect(onPressMock).toHaveBeenCalled();
    });
});
