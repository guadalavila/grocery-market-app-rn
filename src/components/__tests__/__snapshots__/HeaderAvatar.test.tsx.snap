// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HeaderAvatar component should match with snapchot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "space-between",
      },
      Object {
        "marginHorizontal": 16,
        "marginVertical": 24,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Grocery Market
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#68B984",
        "borderRadius": 20,
        "opacity": 1,
        "padding": 8,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "backgroundColor": "#FF1C1E",
          "borderRadius": 18,
          "position": "absolute",
          "right": -6,
          "top": -10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontWeight": "600",
            "paddingHorizontal": 2,
          }
        }
      >
        0
      </Text>
    </View>
    <MockedIoniconsIcons
      color="#FFFFFF"
      name="cart"
      size={22}
      testID="icon-cart"
    />
  </View>
</View>
`;
